modernize(users-portlet): Implement rest api to get user permissions [#33345]

hassandotcms opened the pull request:
### Proposed Changes
* Implemented rest api to fetch user permissions by id/email (by hosts/folder)
* Added integration test, to verify correct permissions combinations for hosts and folders.
* Added postman test


This PR fixes: #33345

This PR fixes: #33345

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- add new response view class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
- implement helper methods for user permission rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
 - get api implementation to return user specific permissions by id/email, by hosts/folder.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- integration test added to verify the functionality of get user permissions rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

add postman test.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

fix integration tests, response type error.

jdotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/user/ResponseEntityUserPermissionsView.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.util.Map;
+
+public class ResponseEntityUserPermissionsView extends ResponseEntityView<Map<String, Object>> {
```
Doc

jdotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserPermissionHelper.java:
```diff
@@ -0,0 +1,262 @@
+            throws DotDataException, DotSecurityException {
+
+        final User systemUser = userAPI.getSystemUser();
+        final Host systemHost = hostAPI.findSystemHost(systemUser, false);
```
this is ApiLocator.systemHost()

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:
```diff
@@ -1334,4 +1354,98 @@ public final void delete(@Context final HttpServletRequest httpServletRequest,
+			final List<Map<String, Object>> permissions = UserPermissionHelper.getInstance()
+				.buildUserPermissionResponse(userRole, requestingUser);
+
+			final Map<String, Object> responseData = Map.of(
```
yeah I see map mostly used across the codebase.
But you are right, view is also a good idea to follow

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:
```diff
@@ -1334,4 +1354,98 @@ public final void delete(@Context final HttpServletRequest httpServletRequest,
+			);
+
+			return Response.ok(new ResponseEntityUserPermissionsView(responseData)).build();
+		} catch (DotDataException e) {
```
I see. was trying to log and throw exception, but I think you are right here

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

support api classes to be available for injection via CDI.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

enable CDI for userPermissionHelper, to enable injection.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 fix integ tests

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

  - Return specific response type instead of generic Response
  - Add better documentation and error messages
  - Split user loading into separate helper method

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 refactoring and cleanup.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

move helper method to helper class.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

wezell reviewed (approved):
This is looking good.  Good permission checks
